model/riscv_types.sail

```diff
@@ -16,6 +16,16 @@
 val extensionEnabled : extension -> bool
 scattered function extensionEnabled
 
+// Priv/unpriv spec version.
+// TODO: Replace the `true`s with configuration flags.
+
+// Machine and Supervisor Architectures v1.11
+function clause extensionEnabled(Ext_Sx1p11) = true
+// Machine and Supervisor Architectures v1.12
+function clause extensionEnabled(Ext_Sx1p12) = extensionEnabled(Ext_Sx1p11) | true
```

Comment:
This doesn't make sense to me. You can't have multiple versions of an extension implemented. Ss1p12 is implemented if and only if priv 1.12 is the version your implementation chose. You have the Ss and Sm extensions and they have a version corresponding to whatever's implemented.

What you're after is some `extensionVersion` function that, given Ss (NB: no suffix on the extension name in the enum), tells you what version is implemented, so you can do `>= 1.12` where needed.

Reply:
Hmm yeah that is definitely the more correct way to do it, but it seemed a lot simpler to pretend they are separate extensions and then ensure that version N implies versions <N.

Open to other options though.

Reply:
Simpler, sure, but definitely wrong, and for the official model that's a real problem.

Reply:
Is it actually *wrong*? If a chip supports Ss1p12 and some instruction requires Ss1p11 then isn't it fair to say the chip supports Ss1p12 and Ss1p11... Unless the versions aren't backwards compatible I guess. Which tbf is the sometimes the case.

Reply:
Ah ok I understand. I'll try and come up with something better.